ucm2/USB-Audio/Realtek/ALC4080-HiFi.conf

```diff
@@ -0,0 +1,57 @@
+SectionDevice."Speaker" {
+	Comment "Speakers"
+	Value {
+		PlaybackChannels 8
+		PlaybackPriority 200
+		PlaybackPCM "hw:${CardId}"
+		JackControl "Speaker - Output Jack"
+		PlaybackMixerElem "Speaker"
+	}
+}
+
+SectionDevice."Headphones" {
+	Comment "Front Headphones"
+	Value {
+		PlaybackPriority 300
+		PlaybackPCM "hw:${CardId},1"
+		JackControl "Headphone - Output Jack"
+		PlaybackMixerElem "Front Headphone"
+	}
+}
+
+SectionDevice."SPDIF" {
+	Comment "S/PDIF Out"
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId},3"
+		PlaybackMixerElem "IEC958"
+	}
+}
+
+SectionDevice."Line" {
+	EnableSequence [
+		cset "name='Line Capture Switch' on"
+	]
+
+	Comment "Line In"
+	Value {
+		CapturePriority 100
+		CapturePCM "hw:${CardId},1"
+		JackControl "Line - Input Jack"
+		CaptureMixerElem "Line"
+	}
+}
+
+SectionDevice."Mic" {
+	EnableSequence [
+		cset "name='Mic Capture Switch' on"
+	]
+
+	Comment "Microphone"
+	Value {
+		CapturePriority 200
+		CapturePCM "hw:${CardId},2"
```

Comment:
This works but only on the front connector. Back connector gets no input. Also it seems (with a typical lavalier microphone) to be low volume. Talking at normal recording volume reaches about -30 dBm and I am expecting to get closer to -12. Maybe not a UCM issue.

Reply:
Front Mic works on MSI MAG X570S Torpedo Max (tested), rear connector not tested; level somewhat low

Reply:
Level is indeed a bit low. The volume levels on Windows are noticeably higher. I guess Windows seems to apply some boost (even without enabling it in the settings).

Back microphone connector definitely works on hw:${CardId},2 on my board. Front and back levels are between -42dBm to -36dBm during normal talking for me.

Reply:
I think you can use `cset` to set the level like here: https://github.com/alsa-project/alsa-ucm-conf/blob/master/ucm2/Qualcomm/apq8096/HiFi.conf#L18= If this value is dBm, then maybe it should be about 20?

Reply:
I tried setting the volume this way and it made no difference.